### PR TITLE
fix(manage-issue-due-dates): don't add assignee ping comment if already exists

### PR DESCRIPTION
**Pull request by k-borley**
### :pencil: Description
- Will prevent this helper from adding a new assignee ping comment (such as `@assignee, this issue assigned to you is now overdue`) if the comment already exists
- Still adds a new comment if the assignee or status (due soon vs. overdue) changes

### CODE REVIEW

File: src/utils/add-due-date-comment.ts [hunk 1]
```diff
@@ -11,17 +11,20 @@ See the License for the specific language governing permissions and
...
-  const hasExistingComments = numComments > 0;
-  const commentList: CommentList = hasExistingComments ? await paginateAllCommentsOnIssue(issue_number) : [];
-  // Create due date comment if there are no existing comments or the comment list does not contain a due date comment
-  if (!hasExistingComments || !commentList.find((comment: SingleComment) => comment.body?.startsWith('This issue is due on'))) {
+export const addDueDateComment = async (deadline: number, createdDate: Date, issue_number: number) => {
+  const commentList: CommentList = await paginateAllCommentsOnIssue(issue_number);
+  if (
+    !commentList ||
+    !commentList.find(
+      (comment: SingleComment) => comment.user?.login === 'github-actions[bot]' && comment.body?.startsWith('This issue is due on')
```
Comment (danadajian) [inline src/utils/add-due-date-comment.ts:25]: It seems presumptuous that the comment would be left by the github-actions service account, what if somebody is using a personal access token to run this workflow?

Comment (k-borley) [reply src/utils/add-due-date-comment.ts:25]: Good point, didn't realize that was an option

File: src/utils/add-due-date-comment.ts [hunk 2]
```diff
@@ -11,17 +11,15 @@ See the License for the specific language governing permissions and
...
 import { SECONDS_IN_A_DAY } from '../constants';
 
-export const addDueDateComment = async (deadline: number, createdDate: Date, issue_number: number, numComments: number) => {
-  const hasExistingComments = numComments > 0;
-  const commentList: CommentList = hasExistingComments ? await paginateAllCommentsOnIssue(issue_number) : [];
-  // Create due date comment if there are no existing comments or the comment list does not contain a due date comment
-  if (!hasExistingComments || !commentList.find((comment: SingleComment) => comment.body?.startsWith('This issue is due on'))) {
+export const addDueDateComment = async (deadline: number, createdDate: Date, issue_number: number) => {
+  const commentList: CommentList = await paginateAllCommentsOnIssue(issue_number);
+  if (!commentList || !commentList.find((comment: SingleComment) => comment.body?.startsWith('This issue is due on'))) {
```
Comment (danadajian) [inline src/utils/add-due-date-comment.ts:22]:
```suggestion
  if (!commentList?.find((comment: SingleComment) => comment.body?.startsWith('This issue is due on'))) {
```

**Pull request merged**